include/particle.tcc

```diff
@@ -208,7 +208,7 @@ bool mpm::Particle<Tdim, Tnphases>::assign_cell_id(mpm::Index id) {
 template <unsigned Tdim, unsigned Tnphases>
 void mpm::Particle<Tdim, Tnphases>::remove_cell() {
   // if a cell is not nullptr
-  if (cell_ != nullptr) cell_->remove_particle_id(this->id_);
+  // if (cell_ != nullptr) cell_->remove_particle_id(this->id_);
```

Comment:
why is this commented out?

Reply:
When iterate through cells, one needs to iterate through local paticles_ids in that cells and then delete them, but when calling this function once, the particles_ids would be deleted and the iteration process would be stopped. So I tried to comment it out.

Reply:
We can write another similar function which comment this out and the original one can remain the same.

Reply:
> why is this commented out?

When iterate through cells, one needs to iterate through local paticles_ids in that cells and then delete them, but when calling this function once, the particles_ids would be deleted and the iteration process would be stopped. So I tried to comment it out.

Reply:
but this is just removing id

Reply:
> why is this commented out?

We can write another similar function which comment this out and the original one can remain the same.